### ISSUE TITLE
Compatibility with new eslint config spec (flat config)

### ISSUE THREAD

**Issue by jjangga0214** [prettier/eslint-config-prettier#231]
Eslint announced a new config spec.
https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new

<img width="879" alt="스크린샷 2022-09-09 오후 10 07 49" src="https://user-images.githubusercontent.com/28584151/189357000-2016d6e8-d655-4299-9bcd-b062c0cedb7f.png">

According to [eslint's official blog](https://eslint.org/blog/2022/08/new-config-system-part-2/), though backward compatibility was taken care of, not all shareable configs continue to work with the new config system.

Is `eslint-config-prettier` compatible with the new config?

Thanks.

**Comment by lydell** [comment]
I tried it out quickly, and in short: Yes, it works. But there are some things to do as well.

eslint.config.js:

```js
import prettier from "eslint-config-prettier";

export default [
    "eslint:recommended",
    prettier,
];
```

The above successfully disables the [no-mixed-spaces-and-tabs](https://eslint.org/docs/latest/rules/no-mixed-spaces-and-tabs) rule from `eslint:recommended`! 🎉 

And this:

```js
import prettier from "eslint-config-prettier";

export default [
    {
        rules: {
            semi: "error",
        }
    },
    prettier,
];
```

The above successfully turns off the `semi` rule. 🎉  Even though `semi` wasn’t set by another config. This wasn’t possible before.

However, the CLI helper tool does not work:

```
❯ npx eslint-config-prettier index.js
No ESLint configuration found in /Users/lydell/stuff/eslint.
```

TODO:

- [ ] Make the CLI helper tool work and make sense with the new config format.
- [ ] Update documentation with examples of both old and new format.
- [ ] Make it clear in the documentation that you need to put `prettier` _after_ things that you want to override, and that it can “win” over not just other configs but also your own config (as mentioned above).

**Comment by lydell** [comment]
Changing this line:

https://github.com/prettier/eslint-config-prettier/blob/a57a3bb0e5aa39f59c2523bb3b77bf7e13b7be2a/bin/cli.js#L11

```diff
-const { ESLint } = require(require.resolve("eslint", {
+const { FlatESLint: ESLint } = require(require.resolve("eslint/use-at-your-own-risk", {
```

… seems to make the CLI work with the new config system. However, that is still an unsupported API, and I need to figure out how to support both the old and new config systems at the same time.

I think the conclusion for now is:

- The main part of eslint-config-prettier works already with the new config system.
- Early adopters can find this issue for information.
- Let’s wait for ESLint to progress more. The above experiment indicates that we should be able to make the CLI work without too much effort when the time comes.

**Comment by theoparis** [comment]
Unless I'm doing something wrong it does not seem to work anymore 😕 
```
Oops! Something went wrong! :(

ESLint: 8.44.0

TypeError: Cannot read properties of undefined (reading 'includes')
    at /home/theo/dev/fem-kanban/node_modules/.pnpm/eslint-plugin-prettier@5.0.0-alpha.2_eslint-config-prettier@8.8.0_eslint@8.44.0_prettier@3.0.0/node_modules/eslint-plugin-prettier/worker.js:116:27

```

**Comment by lydell** [comment]
The updated CLI tool was released in v9.0.0.

The docs and tests have been updated too.
